scripts/update-protos.sh

```diff
@@ -0,0 +1,417 @@
+#!/bin/bash
+
+# Update Protos Script
+# This script handles updating proto submodules, generating Go code, and managing versions
+# Can be invoked by GitHub Actions or run locally
```

Comment:
> or run locally

Unless you use a non-bash platform. SDKs should not limit the user/contributor platforms by requiring bash IMO. I would recommend Go programming language over bash programming language myself, but it's not a big deal.

Reply:
The primary use case I was targeting was via a github action. 
I wasn't planning to spend too much time on this. 

We can look into converting this into a go script when the need arises.